spec/lib/discourse_custom_topic_lists/custom_topic_list_spec.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+RSpec.describe DiscourseCustomTopicLists::CustomTopicList do
+  fab!(:admin)
+  fab!(:user)
+
+  class SiteSettingHelper
+    def self.add_json(*jsons)
+      jsons.each do |json|
+        site_setting = JSON.parse SiteSetting.custom_topic_lists
+        site_setting << json
+        SiteSetting.custom_topic_lists = site_setting.to_json
+      end
+    end
+  end
+
+  before do
+    SiteSetting.discourse_custom_topic_lists_enabled = true
+    SiteSetting.experimental_topics_filter = true
+    SiteSettingHelper.add_json(
+      {
+        "icon" => "question",
+        "name" => "New questions",
+        "description" => "Topic for new questions",
+        "slug" => "some-long-path-for-questions",
+        "query" => "category:questions",
+        "showOnSidebar" => true,
+        "showOnDropdown" => true,
+        "access" => "admins",
+      },
+      {
+        "icon" => "discourse-sparkles",
+        "name" => "Arts and Media",
+        "description" => "## Topic with categories related to arts and media",
+        "slug" => "arts-and-media",
+        "query" => "category:arts-media",
+        "showOnSidebar" => true,
+        "showOnDropdown" => true,
+        "access" => "",
+      },
+    )
+  end
+
+  describe "#lists" do
+    it "returns all custom lists for admins" do
+      custom_lists = described_class.new(admin).lists
+      expect(custom_lists.size).to eq(2)
+    end
+
+    it "returns only lists that the user has access to" do
+      custom_lists = described_class.new(user).lists
+      expect(custom_lists.size).to eq(1)
```

Comment:
```suggestion
      expect(custom_lists.size).to eq(1)
      expect(custom_lists[0]["name"]).to eq("Arts and Media")
```

Asserting for the size only is not sufficient because the method can incorrectly return the `New question` custom list for the given user and this assertion will still pass.